libs/langchain/langchain/chains/base.py

```diff
@@ -148,22 +147,25 @@ def invoke(
         new_arg_supported = inspect.signature(self._call).parameters.get("run_manager")
         run_manager = callback_manager.on_chain_start(
             dumpd(self),
-            inputs,
+            input,
```

Comment:
We need to modify the logic a bit to avoid introducing a breaking change.

Currently, on the happy path all get the output of `prep_inputs` rather than `input`, because those two values are different, we need to make sure that we don't modify the happy batch.

What we can do is something like:

```python
try:
  inputs = self.prep_inputs(input)
except BaseException as e:
  # un-happy path
  invoke on chain start callback handler
  invoke on chain error call back hanlder

# continue with happy path (same code as before)
on_chain_start using the output from self.prep_inputs
```

Reply:
Looks good! Sorry for the delay in reviewing.